### DISCUSSION TITLE
Spack configuration for Frontier

### DISCUSSION THREAD

**Discussion by kmorel** [spack/spack#45607] [Q&A]
Can someone point me to a working spack configuration setup for Frontier? I'm trying to compile some rocm software via spack. I've tried to just clone spack and compile things, but that didn't work. I've tried following [instructions on OLCF's documentation](https://docs.olcf.ornl.gov/software/spack_environments.html), but they seem too out of date. The documentation is for Summit, and swapping out the spack.yaml file for Frontier does not work. I tried to use the [external spack config from E4S](https://github.com/E4S-Project/facility-external-spack-configs/tree/master/frontier). That used to work, but it no longer does. I note that there are no commits for the last 9 months and it seems out of date.

What is the right spack configuration to use?

**Comment by kwryankrattiger** [comment]
I have some local updates to the configs from E4S that I just pushed.

https://github.com/E4S-Project/facility-external-spack-configs/pull/13

**Reply by kmorel** [reply]
Thanks for updating these. However, spack is still giving me a hard time using this configuration.

Based on your previous advice, I am setting the `SPACK_USER_CONFIG_PATH` environment variable to the `frontier` directory of this repo. I am also setting up cache keys for the mirror with `spack buildcache keys --install --trust`. After that, I am creating an environment with the following configuration:

```yaml
spack:
  concretizer:
    reuse: true
  packages:
    all:
      # Use a specific version of GCC
      require: "%gcc@11.2.0"
  include:
  - $spack/../spack-configs/e4s-configs/frontier/rocm-5.3.0
```

I throw in a `spack concretize -f` here as well. However, when I try to install the vtk-m module, I get a bunch of errors about not satisfying requirements.

```
==> Error: failed to concretize `vtk-m@2.1.0+examples+kokkos~openmp+rocm amdgpu_target=gfx90a` for the following reasons:
     1. cannot satisfy a requirement for package 'zlib-ng'.
     2. cannot satisfy a requirement for package 'vtk-m'.
     3. cannot satisfy a requirement for package 'rocthrust'.
     4. cannot satisfy a requirement for package 'pkgconf'.
     5. cannot satisfy a requirement for package 'llvm-amdgpu'.
     6. cannot satisfy a requirement for package 'kokkos'.
     7. cannot satisfy a requirement for package 'hsa-rocr-dev'.
     8. cannot satisfy a requirement for package 'hip'.
     9. cannot satisfy a requirement for package 'gmake'.
    10. cannot satisfy a requirement for package 'glibc'.
    11. cannot satisfy a requirement for package 'curl'.
    12. cannot satisfy a requirement for package 'cmake'.
    13. curl: Only supported on macOS
    14. curl: Only supported on macOS
        required because conflict constraint tls=secure_transport 
        required because conflict is triggered when arch=linux-None-None 
```

I think this is caused by the `%gcc@11.2.0` requirement in my environment configuration. But if I don't have that, my build goes south fast. Any idea how to resolve that?

**Reply by kmorel** [reply]
Never mind on that. Right after I sent this I reset everything and everything started building. We'll see if it gets to the end.

**Comment by kmorel** [comment]
Thanks to @kwryankrattiger's configuration, I was able to set up a configure and build. My setup is documented here: https://github.com/kmorel/vtkm-frontier-scripts.
